lib/actions-oclif/internal/osinit.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license
+ * Copyright 2016-2020 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { stripIndent } from 'common-tags';
+import Command from '../../command';
+import { CommandHelp } from '../../utils/oclif-utils';
+
+// These are internal commands we want to be runnable from the outside.
+// One use-case for this is spawning the minimal operation with root privileges.
+
+interface ArgsDef {
+	image: string;
+	type: string;
+	config: string;
+}
+
+export default class OsinitCmd extends Command {
```

Comment:
Could anyone provide example usage of this command?  (I haven't been able to fully test)

Reply:
It's an internal command, so do a text search of the CLI code for `osinit` and find out how it is being invoked...

Depending on how far you wanted to take the PR / refactoring, you could perhaps ***not*** create the `internal osinit` and `internal scandevices` commands, and instead just refactor the code as modules / functions (rather than commands) and have the callers import the modules and call the function.

Reply:
I see thanks.  I thought internal just meant "for internal use".

Regarding the extended refactor, I think I'll limit this PR to the ts/oclif conversion (not that it's done), and add a comment that this should be done at some point.

Reply:
> I haven't been able to fully test

The occurrence I found for `internal osinit` was in `lib/actions/os.coffee`, in the implementation of the `os initialize` command. For `scandevices`, I found an occurrence in `lib/utils/promote.ts`, used by the `balena join` and `balena leave` commands. Please test that these commands still work with the changes in this PR (I mean testing `os initialize`, `balena join`, `balena leave`).

Reply:
https://www.flowdock.com/app/rulemotion/i-cli/threads/yveDZUzzT4f5g2INiLiD9s83UDJ